kitsune/sumo/middleware.py

```diff
@@ -300,6 +300,10 @@ def process_view(self, request, callback, callback_args, callback_kwargs):
         if 'aaq' in view_name:
             request.session['in-aaq'] = True
             request.session['product_key'] = callback_kwargs.get('product_key')
+        elif 'questions' in request.META.get("HTTP_REFERER", ""):
```

Comment:
This will also include the `/questions` url which is not part of the AAQ. Also the if statement is focused on checking the views and not the request object. How about moving that under the else statement?
```python
...
else:
    request.session['in-aaq'] = False
    if not 'questions/new' in referer:
        reset_key_here
```